bin/tester/patchBuild.js

```diff
@@ -0,0 +1,49 @@
+import { spawn } from 'yarn-or-npm';
+import dedent from 'ts-dedent';
+
+import {
+  isClean,
+  isUpToDate,
+  getUpdateMessage,
+  findMergeBase,
+  checkout,
+  checkoutPrevious,
+} from '../git/git';
+import { runTest } from './index';
+
+const installDependencies = () => spawn.sync(['install'], { stdio: 'inherit' });
+
+export async function runPatchBuild(options) {
+  const [headRef, baseRef] = options.patchBuild;
+
+  // Make sure the git repo is in a clean state (no changes / untracked files).
+  if (!(await isClean())) {
+    throw new Error(dedent`
+      The git working directory must be clean before running a patch build.
+        (use "git stash --include-untracked --keep-index" to stash changes before you continue)
+    `);
+  }
+
+  // Make sure both the head and base branches are up-to-date with the remote.
+  if (!(await isUpToDate())) {
+    throw new Error(await getUpdateMessage());
+  }
+
+  // Get the merge base commit hash.
+  const mergeBase = await findMergeBase();
+  if (!mergeBase) {
+    throw new Error(dedent`
+      Failed to retrieve the merge-base. You may have specified an invalid base branch.
+      Are you sure the head branch is a descendant (i.e. fork) of the base branch?
+        (try running this command yourself: "git merge-base ${headRef} ${baseRef}")
+    `);
+  }
+
+  checkout(mergeBase);
+  installDependencies();
+
+  await runTest({ ...options }); // TODO set branch
```

Comment:
We need to tell the index that this is a patch build so it can do the [follow up steps](https://github.com/chromaui/chromatic/issues/3238#issuecomment-605889033):

 4. We have to trigger a comparison when this patch build is created (normally a base build won't trigger one).
 5. The patch build should be appended to the parent builds of the build on the head branch most recently created after the committedAt date for the patch build.

Note that:

 A. Neither of these steps are possible if the head build doesn't exist yet
 B. Even if the head build exists, we still need to know the PR number in order to find the head build & PR (just the branch alone isn't enough as you can have more than one PR per branch -- or is it possible to tell from the combination of `headRef` & `baseRef`?)

Given A), I think we need to either: 

 i. bail if the head build doesn't exist yet (otherwise there's little point in all this) OR
 ii. create the head build first. OR
 iii. create the head build second *but* tell it the "merge base to use" somehow.

Given B) (unless I am wrong about telling the PR from the combination), I think we need to *pass the PR number* with this command, and look up the `baseRef`/`headRef` on the index.


I would suggest we maybe start with Ai. for now (which will require looking up the head build on the index) and in our instructions tell the user to run a build on the head branch *first*?

WDYT?